File: pkg/dev_compiler/test/codegen_expected/lib/html/client_rect_test.js
dart_library.library('lib/html/client_rect_test', null, /* Imports */[
  'dart_sdk',
  'unittest'
], function load__client_rect_test(exports, dart_sdk, unittest) {
  'use strict';
  const core = dart_sdk.core;
  const math = dart_sdk.math;
  const html = dart_sdk.html;
  const dart = dart_sdk.dart;
  const dartx = dart_sdk.dartx;
  const src__matcher__core_matchers = unittest.src__matcher__core_matchers;
  const html_config = unittest.html_config;
  const unittest$ = unittest.unittest;
  const src__matcher__expect = unittest.src__matcher__expect;
  const client_rect_test = Object.create(null);
  let RectangleOfnum = () => (RectangleOfnum = dart.constFn(math.Rectangle$(core.num)))();
  let ListOfRectangleOfnum = () => (ListOfRectangleOfnum = dart.constFn(core.List$(RectangleOfnum())))();
  let dynamicTobool = () => (dynamicTobool = dart.constFn(dart.definiteFunctionType(core.bool, [dart.dynamic])))();
  let VoidToElement = () => (VoidToElement = dart.constFn(dart.definiteFunctionType(html.Element, [])))();
  let VoidTodynamic = () => (VoidTodynamic = dart.constFn(dart.definiteFunctionType(dart.dynamic, [])))();
  client_rect_test.main = function() {
    let isRectList = src__matcher__core_matchers.predicate(dart.fn(x => ListOfRectangleOfnum().is(x), dynamicTobool()), 'is a List<Rectangle>');
    function insertTestDiv() {
      let element = html.Element.tag('div');
      element[dartx.innerHtml] = '    A large block of text should go here. Click this\n    block of text multiple times to see each line\n    highlight with every click of the mouse button.\n    ';
      html.document[dartx.body][dartx.append](element);
      return element;
    }
    dart.fn(insertTestDiv, VoidToElement());
    html_config.useHtmlConfiguration();
    unittest$.test("ClientRectList test", dart.fn(() => {
      insertTestDiv();
      let range = html.Range.new();
      let rects = range[dartx.getClientRects]();
      src__matcher__expect.expect(rects, isRectList);
    }, VoidTodynamic()));
  };
  dart.fn(client_rect_test.main, VoidTodynamic());
  // Exports:
  exports.client_rect_test = client_rect_test;
});

File: tests/lib_2/html/client_rect_test.dart
import 'dart:html';

import 'package:expect/minitest.dart';

main() {
  var isRectList =
      predicate((x) => x is List<Rectangle>, 'is a List<Rectangle>');

  insertTestDiv() {
    var element = new Element.tag('div');
    element.innerHtml = r'''
    A large block of text should go here. Click this
    block of text multiple times to see each line
    highlight with every click of the mouse button.
    ''';
    document.body.append(element);
    return element;
  }

  test("ClientRectList test", () {
    insertTestDiv();
    var range = new Range();
    var rects = range.getClientRects();
    expect(rects, isRectList);
  });

  test("ClientRect ==", () {
    var rect1 = document.body.getBoundingClientRect();
    var rect2 = document.body.getBoundingClientRect();
    expect(rect1, equals(rect2));
  });
}
